src/group_map.rs

```diff
@@ -15,7 +15,7 @@ pub fn into_group_map<I, K, V>(iter: I) -> HashMap<K, Vec<V>>
     let mut lookup = HashMap::new();
 
     for (key, val) in iter {
-        lookup.entry(key).or_insert(Vec::new()).push(val);
+        lookup.entry(key).or_insert_with(Vec::new).push(val);
```

Comment:
`Vec::new` is *extremely* cheap, and shouldn't be thunked. This almost makes me want to file a clippy bug...

Reply:
Hmm, can clippy tell whether it's `const fn`?  Would be nice to disable that link if the RHS is a const.

Reply:
@jswrenn I agree that we should avoid unnecessary work for the compiler where possible. However, thunking a `Vec::new` has (significant?) run-time advantages [in debug mode](https://godbolt.org/z/jfjnv9)...  maybe nice to have.

[Release mode](https://godbolt.org/z/TqYvTE) brings them closer together (possibly equivalent in run-time).